### PR TITLE
wicked: Improve upload and tcpdump handling

### PR DESCRIPTION
**Pull request by okurz**
* wicked: Be strict about tcpdump presence to prevent alarm fatigue
* wicked: Ensure tcpdump is also installed for 'ref'
* serial_terminal: Fix grammar in die-message

Verification runs:
* ref: https://openqa.suse.de/tests/3552959
* sut: https://openqa.suse.de/tests/3552960

### CODE REVIEW

**Comment by asmorodskyi** [comment]
I thought we have agreed on something different . Fully ignoring seems to me better choice. Last thing which I want to have is wicked test is failing because there is no tcpdump available in repos

File: lib/serial_terminal.pm [hunk 1]
```diff
@@ -178,7 +178,7 @@ sub upload_file {
     $dst = "ulogs/" . $dst;
     my ($fh, $tmpfilename) = tempfile(UNLINK => 1, SUFFIX => '.openqa.upload');
 
-    die("File $src doesn't exists on SUT") if (script_run("test -f $src", quiet => 1, timeout => $opts{timeout}) != 0);
+    die("File $src does not exist on SUT") if (script_run("test -f $src", quiet => 1, timeout => $opts{timeout}) != 0);
```
Comment (asmorodskyi) [inline lib/serial_terminal.pm]: why you say that it is wrong ? https://www.englisch-hilfen.de/en/grammar/kurzformen.htm short forms is grammatically correct versions of long one

Comment (okurz) [reply lib/serial_terminal.pm]: the combination of "does not exist**s**" would be wrong. I just fixed it *and* avoided the contraction which is a common good style for formal written english

**Comment by okurz** [comment]
> please remove behavior which would cause wicked test to die in case anything related to tcpdump goes wrong

that would mean I would need to handle every call to tcpdump conditionally in a way to also foresee the possibility that it was never installed, right?

**Comment by asmorodskyi** [comment]
> > please remove behavior which would cause wicked test to die in case anything related to tcpdump goes wrong
> 
> that would mean I would need to handle every call to tcpdump conditionally in a way to also foresee the possibility that it was never installed, right?

taking into account condition **wicked tests should not failed in case any issue with tcpdump** I am wondering about what kind of different calls for tcpdump we speaking here ? Currently all calls to tcpdump are done in `script_run` so no matter if they pass or fail we moving forward which fully apply to condition which I mentioned so what kind of improvement could be done here ?

**Comment by asmorodskyi** [comment]
Your main concern was "red squares in test which lead to alarm fatigue"  , so only way how to solve your concern and keep my condition is to lower down `record_info` from fatal to info . Don't you mind for having info box about issues with tcpdump ?

**Comment by asmorodskyi** [comment]
also I am fine with grammar fix and adding tcpdump for ref

**Comment by okurz** [comment]
@asmorodskyi as you stated that you are fine with 2/3 commits I have removed the last one for now so we should be good to merge as is now. Regarding the "strictness": It is not just about "red boxes", much more about failed commands which are ignored. E.g. the "kill" command to terminate the tcpdump process fails with a confusing message because the PID does not relate to a still running process anymore. So this is not about arguing between "very confusing" and "much confusing" ;)

**Comment by jlausuch** [comment]
I think we are making a big deal of this.

Tcpdump is not mandatory for wicked tests to pass, just a plus to collect more logs for further troubleshooting. So, it's ok to use script_run for this command and let the test pass if the command fails. Besides, if do a check for tcpdump command, following the same philosophy we would need to do the same for many other commands we are running, which doesn't make sense.

I am not saying we don't care if the command fails... We for sure will investigate why and fix it when this type of things happen (which is what we did).

**Comment by okurz** [comment]
@jlausuch don't worry, asmorodskyi and me like to argue about general concepts based on specific examples. It will stay civilized, trust me :)

The problem I encountered is that I was checking if a worker machine is ready to bring back into production and saw a lot of problems, e.g. a call to "kill" failing to find the tcpdump process which could have never started because the executable is not there. Similar things I have encountered especially within the wicked tests, unhandled errors, false positives, red herrings, all not influencing the final job result because it did not actually show wicked errors but still very misleading. If you do not want me to change this at all the only other viable alternative I see is that whenever something with "wicked" fails I would assume it's not a real problem and leave it to you to proof that there are not wicked-related changes that lead to failures, e.g. differing worker setup.

**Pull request merged**